src/coap_client.c

```diff
@@ -421,10 +448,19 @@ static enum golioth_status golioth_coap_client_get_internal(struct golioth_clien
 
     if (is_synchronous)
     {
-        // Created here, deleted by coap thread (or here if fail to enqueue
+        // Created here, deleted by coap thread (or here if fail to create or enqueue)
         request_msg.request_complete_event = golioth_event_group_create();
         request_msg.request_complete_ack_sem = golioth_sys_sem_create(1, 0);
         request_msg.status = &status;
+
+        if (!request_msg.request_complete_event || !request_msg.request_complete_ack_sem)
+        {
+            GLTH_LOGW(TAG, "Failed to create request_complete members");
+            golioth_event_group_destroy(request_msg.request_complete_event);
+            golioth_sys_sem_destroy(request_msg.request_complete_ack_sem);
```

Comment:
We can't assume that `golioth_sys_sem_destroy()` is tolerant to being called with a semaphore that did not create successfully. For example. FreeRTOS will assert if you attempt to destroy a NULL semaphore, and the ESP-IDF port does not check for NULL.

We should just check each one of these for failure after creating it. There's not an advantage in checking them together other than reducing the lines of code, but that's not an advantage in itself and in this case I believe having separate checks is more readable and maintainable.